### ISSUE TITLE
C++ enums being defined in multiple files after b265433 (multi-thread input processing)

### ISSUE THREAD

**Issue by swt2c** [doxygen/doxygen#7979]
**Describe the bug**
After upgrading to 1.8.20, I observed a problem where enums are being defined in multiple files (in addition to the one where they are actually defined).  I bisected this behavior to b265433.  This is the behaviour with b265433:
```
event_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
filename_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
headercol_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
index.xml:    <member refid="interface_2wx_2listctrl_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="event_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="filename_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="headercol_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="webkit_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="toplevel_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="regex_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
index.xml:    <member refid="utils_8h_1adf764cbdea00d65edcd07bb9953ad2b7ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
interface_2wx_2listctrl_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
regex_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
toplevel_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
utils_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
webkit_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
```

**Expected behavior**
This is the behaviour before b265433:
```
index.xml:    <member refid="webkit_8h_1aac34dfe6c6b73b43a4656c9dce041034ae1379fdf0911b272f03e8dc06c88b463" kind="enumvalue"><name>wxWEBKIT_NAV_LINK_CLICKED</name></member>
webkit_8h.xml:          <name>wxWEBKIT_NAV_LINK_CLICKED</name>
```
The enum value `wxWEBKIT_NAV_LINK_CLICKED` is only defined in webkit.h, so it should only show up in *that* XML file.

**Version**
Running on Fedora Linux 32.  Compiled doxygen myself to bisect.

**Comment by swt2c** [comment]
I should clarify that these are unnamed enums, e.g.,:
```
enum {
    wxWEBKIT_NAV_LINK_CLICKED = 1,
    wxWEBKIT_NAV_BACK_NEXT = 2,
    wxWEBKIT_NAV_FORM_SUBMITTED = 4,
    wxWEBKIT_NAV_RELOAD = 8,
    wxWEBKIT_NAV_FORM_RESUBMITTED = 16,
    wxWEBKIT_NAV_OTHER = 32

};
```

It appears to be that in b265433 and beyond that all unnamed enum values (from multiple declariations) are being combined into a single unnamed enum during the parsing.

**Comment by albert-github** [comment]
I tried to reproduce this and after a number of tries I was successful (anonymous things are always tricky), in future:
> please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

It looks like the anonymous enums are all put on one heap.
The output of the 1.8.18 version:
![ok_1_8_18](https://user-images.githubusercontent.com/5223533/91425376-57bea000-e85b-11ea-98a5-4408f276237e.jpg)

The output of the 1.8.20 (and also of the current master63dc5b9b1b3e8fb875304a954e4df934b249034b of August 26, 2020) version:
![err_1_8_20](https://user-images.githubusercontent.com/5223533/91425389-5beabd80-e85b-11ea-96c9-d2eae18cceac.jpg)

The project I used: [example.tar.gz](https://github.com/doxygen/doxygen/files/5135609/example.tar.gz)

**Comment by swt2c** [comment]
Thanks @albert-github - I was working on a SCCCE, but hadn't been successful in creating one yet.  Thank you!

**Comment by doxygen** [comment]
Please verify if the commit fixes the problem. Don't close the bug report yet.

**Comment by swt2c** [comment]
I can confirm this fixes the issue.  :+1: Thanks a lot for the quick fix!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
